pyOCD/gdbserver/gdbserver.py

```diff
@@ -329,12 +338,27 @@ def run(self):
                     break
 
                 if self.packet_io.interrupt_event.isSet():
-                    logging.debug("Got unexpected ctrl-c, ignoring")
+                    if self.non_stop:
```

Comment:
The [non-stop mode docs](https://sourceware.org/gdb/current/onlinedocs/gdb/Non_002dStop-Mode.html#Non_002dStop-Mode) say that ctrl-c should only apply to the current thread in non-stop mode. But that may just be a UI behaviour, and it might send vCont to the server stub.